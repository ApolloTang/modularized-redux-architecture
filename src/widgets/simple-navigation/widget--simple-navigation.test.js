import React from 'react';
import SimpleNavigation from './index';
import {shallow } from 'enzyme';
import {shallowToJson } from 'enzyme-to-json';


const navigationDirective = [
    {to:'/', displayText:'Home'},
    {to:'/users', displayText:'User'},
    {to:'/users/1', displayText:'User 1'},
    {to:'/users/1/view-review/2', displayText:'User 1 view user 2'},
    {to:'/font-test', displayText:'Font test'},
    {to:'/lazy-load', displayText:'Lazy load'},
    {to:'/todos/all', displayText:'Todo'},
    {to:'/rest', displayText:'Rest'}
];


const navLenght = navigationDirective.lenght;

describe(`
  =============================================
  ==               Widget Test               ==
  ==           simple-navigation             ==
  =============================================
`, () => {
  describe(':::: <SimpleNavigation /> ', () => {
    const component = shallow(<SimpleNavigation navigations={navigationDirective} />);
    const componentJson = shallowToJson(component);

    test(`:::: lenght of children should be one `, () =>{
      expect(component).toHaveLength(1);
    });

    test(`:::: snapshot test `, () =>{
      expect(component).toMatchSnapshot(1);
    });
  })
});


