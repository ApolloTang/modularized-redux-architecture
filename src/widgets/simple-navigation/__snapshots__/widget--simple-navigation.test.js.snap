// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `
<div
  className="simple-navigation module-style"
>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/users"
      >
        User
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/users/1"
      >
        User 1
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/users/1/view-review/2"
      >
        User 1 view user 2
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/font-test"
      >
        Font test
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/lazy-load"
      >
        Lazy load
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/todos/all"
      >
        Todo
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/rest"
      >
        Rest
      </NavLink>
    </li>
  </ul>
</div>
`;
